src/Runtime/OMTensorList.inc

```diff
@@ -26,37 +26,33 @@
 #endif
 
 #include "onnx-mlir/Runtime/OMTensorList.h"
+#include <string.h>
 
 struct OMTensorList {
 #ifdef __cplusplus
-  /**
-   * Constructor
-   *
-   * Create an OMTensorList with specified OMTensor pointer array
-   * and the size of the array
-   */
-  OMTensorList(OMTensor *omts[], int64_t n) : _omts(omts), _size(n) {
-    _owning = false;
-  };
 
   /**
    * Constructor
    *
-   * Create an empty OMTensorList for internal API calls.
+   * Create an OMTensorList with specified OMTensor pointer array, the size of
+   * the array and indication of whether or not to perform a shallow destroy of
+   * the OMTensorLisr
```

Comment:
I am confused.... this statement you made is not clear to me. "If delete is used". Delete is not used...so not following.

Reply:
In C++ you can do either `delete list` which then uses the C++ destructor (and if I recall correctly this also happens implicitly when a stack object goes out of scope in C++). Or you can call the "explicit" functions we created `OMTensorListDestory(list)` or `OMTensorListDestoryShallow(list)`.

Reply:
Okay I understand but I think the wording is not clear to me personally. I can emphasis the point of using `omTensorListCreateWithShallow` or setting the `shallow` flag  is to coincide with using `omTensorListDestroyShallow`

Reply:
From my understanding we would use delete as Gong mentioned only when we subsequently use the new keyword. I did a search but maybe others can clarify this or not but are we using the `delete` anywhere in the repo. I do not see that...is this something we advertise. I thought the point of creating the `destroyers` is to explicitly use them.

Reply:
> I did a search but maybe others can clarify this or not but are we using the delete anywhere in the repo.

That's one case where it's called. I believe it can also be called if you create the object on the stack. So for example: (based on https://www.ibm.com/docs/en/zos/2.1.0?topic=only-destructors and https://learn.microsoft.com/en-us/cpp/cpp/destructors-cpp?view=msvc-170)
```
int main () {
     OMTensorList  list = OMTensorList(<args required by constructor) // <Note list isn't a ptr so the obj is on the stack and not the heap>
      // Destructor ~OMTensorList is called before
      // control returns from main()
    }
```
Would implicitly call the C++ OMTensorList destructor without explicit `delete` call

> I thought the point of creating the destroyers is to explicitly use them.

Very much agreed. Overall supporting this is more a side effect of enabling both C and C++. In reality, I don't think anyone will use a stack object or `new`. We'd much rather them use our explicit APIs (the Create and Destroy methods).

Reply:
Okay I tried to compromise and make it more clear.

Reply:
Updated